pkg/kv/kvserver/replica_destroy.go

```diff
@@ -221,19 +222,51 @@ func (r *Replica) disconnectReplicationRaftMuLocked(ctx context.Context) {
 // ID that it hasn't yet received a RangeDescriptor for if it receives raft
 // requests for that replica ID (as seen in #14231).
 func (r *Replica) setTombstoneKey(
-	ctx context.Context, writer storage.Writer, externalNextReplicaID roachpb.ReplicaID,
+	ctx context.Context,
+	reader storage.Reader,
+	writer storage.Writer,
+	externalNextReplicaID roachpb.ReplicaID,
```

Comment:
Since this is now **the** new `NextReplicaID`, consider naming it simpler, like `nextReplicaID`.
Upd: I see this is done further.